packaging/building-native-packages-locally.md

```diff
@@ -0,0 +1,88 @@
+**QEMU user-mode emulation**. This requires a Linux kernel with binfmt_misc support which all modern distributions provide
+this out of the box (support for WSL and Docker Desktop unknown).
+
+1. To set up the required QEMU user-mode emulation: 
+
+   ```bash
+   docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+   ```
+   The QEMU user-mode emulator will run until you reboot.
+
+2. Build the package by running the emulator and specifying the `--platform` option. The current list of architectures we build for, and the correct
+value for the `--platform` option is:
+
+   * 32-bit ARMv7: `linux/arm/v7`
+   * 64-bit ARMv8: `linux/arm64/v8`
+   * 32-bit x86: `linux/i386`
+   * 64-bit x86: `linux/amd64`
+
+   ```bash
+   docker run --privileged multiarch/qemu-user-static --platform VALUE
+   ```
+
+## If the build failed: Finding build logs
+
+You can find build logs and artifacts in the build directory, whose location varies by distribution:
+
+* DEB systems (Ubuntu and Debian) store the build directory inside the container located at `/usr/src/netdata`
+* RPM systems except openSUSE store the build directory inside the container located under `/root/rpmbuild/BUILD/`. The location varies based on the package version number.
+* openSUSE stores the build directory inside the container located under `/usr/src/packages/BUILD`. The location varies based
+on the package version number.
+
+
+## What happens when you run the Netdata package builder?
+
+Netdata packages are built in fully self-contained Docker images (see [Dockerfiles](https://github.com/netdata/helper-images/tree/master/package-builders)). These are published on Docker
+Hub with the image name [`netdata/package-builders`](https://hub.docker.com/r/netdata/package-builders), and tagged using the name and version of the distribution (with the tag corresponding to the suffix on the associated Dockerfile).
+
+The build code has two specific requirements when run:
+
+* It expects the source tree it should build from to be located at `/netdata`, and expects that said source tree
+  is clean (no artifacts left over from previous builds).
+* It expects an environment variable named `VERSION` to be defined, and uses this to control what version number
+  will be shown in the package metadata and filenames.
+
+When you run the package builder the following happens: 
+1. **Preparation**: The source tree gets copied to a temporary location for the build process. This allows the system to mount the souce tree directly from the host without worrying about leaving a dirty tree behind. Then the script creates any templates or moves files as needed to build the package.
+2. **Building**: The package build command is invoked with the correct arguments and the package is built.
+3. **Permanent storage**: The resulting packages are copied to the `artifacts/` directory in the original source tree so they are
+accessible after the container exits.
```

Comment:
I really don’t know that we need the ‘step-by-step’ organization here. It feels out of place IMO given that we are not instructing the user to do anything

Reply:
I disagree to an extent. We are explaining a workflow that is processed in a certain order, so making these different phases more visible helps IMO. 

However, we can debate if it's really necessary to use an ordered list. 
We could go for unordered or plain paragraphs if you prefer.

Reply:
Hmm, I think the issue here that makes it feel so out of place to me is that it feels like prosaic creative writing the way it’s presented here, not technical documentation. IOW, the structure makes it sound to me less like we’re trying to unambiguously explain a process, and more like we’re trying to make the process sound ‘cool’, and possibly like we’re trying to cater to end users (who are not the target audience for this documentation).

Reply:
@DShreve2 What's your take here?